extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/package-info.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+@EverythingIsNonnullByDefault
```

Comment:
Is the quantiles package ready for this? (Everything nullable is annotated?)

Similar question for other packages. On most of the additions of EverythingIsNonnullByDefault I am wondering if the packages in question are ready. Right now, their parameters and return values are all treated as potentially nullable, which seems like the 'safe' bet if the annotations aren't there yet.

Reply:
Without this annotation, IntelliJ treats all method parameters that are not explicitly annotated `@Nullable` as "unknown" (*not* nullable!) (equivalent of [platform types](https://kotlinlang.org/docs/reference/java-interop.html#null-safety-and-platform-types) in Kotlin), and it doesn't warn you about anything, whether you treat them as nullable or not on the caller side or on the method body side.

So without `@EverythingIsNonnullByDefault`, there are two types of method parameters in the package: nullable (those that are annotated `@Nullable`), and "unknown" (those that are not annotated). You could also make parameters non-nullable by explicitly annotating them `@Nonnull`, but we never do this by convention.

With `@EverythingIsNonnullByDefault`, all "unknown" parameters are considered to be non-nullable, and IntelliJ starts to warn you about
 1) Nullable treatment on the method implementation side
 2) Passing provably nullable objects as the corresponding argument on the method calling side
 3) When the method overrides a method in a supertype, where the corresponding parameter is annotated `@Nullable`.

All those warnings are often false-positives, when the parameter is, in fact, nullable, but just wasn't annotated `@Nullable`. This was IntelliJ pushes you to propagate usage of `@Nullable` in the codebase.

So in fact according to our conventions all packages should be annotated `@EverythingIsNonnullByDefault`, I just never took time to do this in a separate PR.

Reply:
Is there an easier way to get this behavior than adding a hundred `package-info.java` files?

Reply:
Good question. There is no such configuration in IntelliJ currently as far as I can tell. https://youtrack.jetbrains.com/issue/IDEA-197110